### ISSUE TITLE
Crash when spawning 132k entities with 9 components

### ISSUE THREAD

**Issue by deathbeam** [BeanCheeseBurrito/Flecs.NET#49]
Not sure if this is binding issue or flecs issue, the coredump is also quite useless:

```
* thread #1, name = 'Game', stop reason = signal SIGSEGV: address not mapped to object
  * frame #0: 0x000075d4dd47abd9 libcoreclr.so`___lldb_unnamed_symbol9858 + 249
    frame #1: 0x000075d4dd3ebd7e libcoreclr.so`___lldb_unnamed_symbol8317 + 126
    frame #2: 0x000075d4dd3ecee2 libcoreclr.so`___lldb_unnamed_symbol8333 + 514
    frame #3: 0x000075d4dd3edcf4 libcoreclr.so`___lldb_unnamed_symbol8341 + 1412
    frame #4: 0x000075d4dd3e7596 libcoreclr.so`___lldb_unnamed_symbol8272 + 374
    frame #5: 0x000075d4dd3ef647 libcoreclr.so`___lldb_unnamed_symbol8375 + 359
    frame #6: 0x000075d4dd46a90a libcoreclr.so`___lldb_unnamed_symbol9794 + 810
    frame #7: 0x000075d4dd469392 libcoreclr.so`___lldb_unnamed_symbol9791 + 242
    frame #8: 0x000075d4dd48e8fc libcoreclr.so`___lldb_unnamed_symbol9968 + 220
    frame #9: 0x000075d4dd319f34 libcoreclr.so`___lldb_unnamed_symbol5639 + 116
    frame #10: 0x000075d4dd616dde libcoreclr.so`___lldb_unnamed_symbol16085 + 510
    frame #11: 0x000075d4dd8a339d libc.so.6`___lldb_unnamed_symbol3666 + 941
    frame #12: 0x000075d4dd92849c libc.so.6`___lldb_unnamed_symbol4097 + 7
```

**Comment by deathbeam** [comment]
Found something:

```
Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
   at Systems.<>c.<ProcessOccluders>b__6_0(Iter it, Field`1 positions, Field`1 occluders, Field`1 translates) in /home/deathbeam/git/raylib-game/src/Systems.cs:line 211
   at Flecs.NET.Core.Invoker.Iter[T0,T1,T2](Iter it, IterFieldCallback`3 callback)
   at Flecs.NET.Core.BindingContext.Functions`3.IterFieldCallbackDelegate(ecs_iter_t* iter)
   at Flecs.NET.Bindings.flecs.ecs_progress(ecs_world_t* world, Single delta_time)
   at Game.Main() in /home/deathbeam/git/raylib-game/src/Game.cs:line 135
[1]    1102445 IOT instruction (core dumped)  ./bin/Release/net8.0/linux-x64/publish/Game
```

When i commented out the access to Occluders it crashed elsewhere. 

This is my Occluder component:

```
public record struct Occluder(Vector2[] Points);
```

Most of my components have simple values but Occluder is first one it encountered with Vec2 array so I assume its getting yeeted somewhere.

**Comment by deathbeam** [comment]
Yea when I changed all my types to unmanaged only I can continue spawning entities easily:

```csharp
public record struct Texture(FixedString64 Path)
{
    public Texture(string path) : this(new FixedString64(path)) { }
}

public record struct Occluder(NativeArray<Vector2> Points)
{
    public Occluder(Vector2[] points) : this(new NativeArray<Vector2>(points.Length))
    {
        for (int i = 0; i < points.Length; i++)
        {
            Points[i] = points[i];
        }
    }
}

public record struct Collider(NativeArray<Vector2> Points)
{
    public Collider(Vector2[] points) : this(new NativeArray<Vector2>(points.Length))
    {
        for (int i = 0; i < points.Length; i++)
        {
            Points[i] = points[i];
        }
    }
}

public record struct GridPath(int Progress, NativeArray<Vector2> Points)
{
    public GridPath(int progress, Vector2[] points) : this(progress, new NativeArray<Vector2>(points.Length))
    {
        for (int i = 0; i < points.Length; i++)
        {
            Points[i] = points[i];
        }
    }
}
```

So its definitely issue with how managed types are being handled.

**Comment by BeanCheeseBurrito** [comment]
> Found something:
> 
> ```
> Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
>    at Systems.<>c.<ProcessOccluders>b__6_0(Iter it, Field`1 positions, Field`1 occluders, Field`1 translates) in /home/deathbeam/git/raylib-game/src/Systems.cs:line 211
>    at Flecs.NET.Core.Invoker.Iter[T0,T1,T2](Iter it, IterFieldCallback`3 callback)
>    at Flecs.NET.Core.BindingContext.Functions`3.IterFieldCallbackDelegate(ecs_iter_t* iter)
>    at Flecs.NET.Bindings.flecs.ecs_progress(ecs_world_t* world, Single delta_time)
>    at Game.Main() in /home/deathbeam/git/raylib-game/src/Game.cs:line 135
> [1]    1102445 IOT instruction (core dumped)  ./bin/Release/net8.0/linux-x64/publish/Game
> ```
> 
> When i commented out the access to Occluders it crashed elsewhere.
> 
> This is my Occluder component:
> 
> ```
> public record struct Occluder(Vector2[] Points);
> ```
> 
> Most of my components have simple values but Occluder is first one it encountered with Vec2 array so I assume its getting yeeted somewhere.

Can you provide a repro that I can test locally? This usually means that you have a null reference somewhere which can possibly happen if you're doing something like this:
```cs
using World world = World.Create();

world.Entity().Add<string>();              // This defaults to a null reference. Will throw exception if you try to access it.
world.Entity().Set<string>("Hello World"); // You need to provide a reference to a managed component with .Set
```

**Comment by deathbeam** [comment]
Yea I checked if I have nulls anywhere but its not possible, at least not explicitely. Here is minimal repro where I managed to reproduce it super easily:

Works at 100k, crashes at 150k. Without the 2 Vector2[] dummies it crashes at 1.6M. Without dummies and string even 10M works fine.

```
using System.Numerics;
using Flecs.NET.Core;

public record struct Texture(string Path);

public record struct Position(float X, float Y);

public record struct Color(byte R, byte G, byte B, byte A);

public record struct Velocity(float X, float Y);

public record struct Destination(float X, float Y);

public record struct Source(int X, int Y, int Width, int Height);

public record struct Translate(float X, float Y);

public record struct Dummy1(params Vector2[] Points);

public record struct Dummy2(params Vector2[] Points);



class Program
{
    private static readonly int winWidth = 1280;
    private static readonly int winHeight = 720;
    private static readonly int numEntities = 150000;
    private static readonly string entityTexture = "assets/enemies/Enemy_010.png";
    private static readonly Random random = new Random();

    private static Vector2 randomVector2(int maxX, int maxY)
    {
        return new Vector2 { X = random.Next(0, maxX), Y = random.Next(0, maxY) };
    }

    private static Color randomColor()
    {
        return new Color
        {
            R = (byte)random.Next(0, 256),
            G = (byte)random.Next(0, 256),
            B = (byte)random.Next(0, 256),
            A = 255,
        };
    }

    private static void createEntity(World world)
    {
        var pos = randomVector2(winWidth, winHeight);
        var color = randomColor();

        world.Entity()
            .Set(new Texture(entityTexture))
            .Set(new Position(pos.X, pos.Y))
            .Set(new Source(0, 0, 24, 24))
            .Set(new Destination(0, 0))
            .Set(new Velocity(0, 0))
            .Set(new Translate(-12, -6))
            .Set(color)
            .Set(new Dummy1(new Vector2(0, 0), new Vector2(0, 0), new Vector2(0, 0), new Vector2(0, 0)))
            .Set(new Dummy2(new Vector2(0, 0), new Vector2(0, 0), new Vector2(0, 0), new Vector2(0, 0)));
    }

    static void Main(string[] args)
    {
        var world = World.Create();

        for (int i = 0; i < numEntities; i++)
        {
            createEntity(world);
        }

        var system = world.System<Texture, Position, Source, Destination, Color>("RenderSystem")
            .Each((ref Texture texture, ref Position position, ref Source source, ref Destination destination, ref Color color) =>
            {
                Console.WriteLine($"Rendering {texture.Path} at {position.X}, {position.Y} with color {color.R}, {color.G}, {color.B}, {color.A}");
            });

        while (world.Progress())
        {
        }
    }
}
```

**Comment by deathbeam** [comment]
Fixed by #54, closing
